hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/SecurityConfig.java

```diff
@@ -330,6 +330,9 @@ public Duration getMaxCertificateDuration() {
     return this.maxCertDuration;
   }
 
+  /**
+   * Whether to require short-lived tokens for block and container operations.
+   */
   public boolean isBlockTokenEnabled() {
```

Comment:
Block token are container token are all validated on datanodes but issued by different services, can we define separate  configurations to enable/disable them differently?

Reply:
I don't think introducing a new config property for this adds value.  I'm not sure why we even have a config property for enabling block tokens.  I think they should have been enabled in all secure clusters, ie. by `ozone.security.enabled=true`.  But now we have `hdds.block.token.enabled`.  Do we really need another config property just to force all users who want security to set this new config, too?

We would also need another one for container token token lifetime.

Reply:
Technical wise, I think that we could live without a separate config. But there are several reasons for having the token on datanode optional:
1. performance: the additional validation adds the cost for each block I/O
2. backward compatibility: old client will be broken without upgrade.
3. deployment with flexibility  

bq. We would also need another one for container token token lifetime.
We can use a default one for the short lived tokens if customer does not need to specify. This is common in  hadoop delegation tokens.